spec/core/data_structures.md

```diff
@@ -4,195 +4,239 @@ Here we describe the data structures in the Tendermint blockchain and the rules
+|------------------|---------------------------------|---------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| Type             | [SignedMsgType](#signedmsgtype) | Either prevote or precommit. [SignedMsgType](#signedmsgtype)                                                             | A Vote is valid if its corresponding fields are included in the enum [signedMsgType](#signedmsgtype) |
+| Height           | int64                           | Height for which this vote was created for                                                  | Must be > 0                                                                                          |
+| Round            | int32                           | Round that the commit corresponds to.                                                       | Must be > 0                                                                                          |
+| BlockID          | [BlockID](#blockid)             | The blockID of the corresponding block.                                                     | [BlockID](#blockid)                                                                                  |
+| Timestamp        | [Time](#Time)                   | Timestamp represents the time at which a validator signed.                                  | [Time](#time)                                                                                        |
+| ValidatorAddress | slice of bytes (`[]byte`)       | Address of the validator                                                                    | Length must be equal to 20                                                                           |
+| ValidatorIndex   | int32                           | Index at a specific block height that corresponds to the Index of the validator in the set. | must be > 0                                                                                          |
+| Signature        | slice of bytes (`[]byte`)       | Signature by the validator if they participated in consensus for the associated bock.       | Length of signature must be > 0 and < 64                                                             |
 
-```protobuf
+## SignedMsgType
+
+Signed message type represents a signed messages in consensus.
+
+```proto
 enum SignedMsgType {
+
   SIGNED_MSG_TYPE_UNKNOWN = 0;
   // Votes
-  PREVOTE_TYPE   = 1;
-  PRECOMMIT_TYPE = 2;
-  // Proposals
-  PROPOSAL_TYPE = 32;
+  SIGNED_MSG_TYPE_PREVOTE   = 1;
+  SIGNED_MSG_TYPE_PRECOMMIT = 2;
+
+  // Proposal
+  SIGNED_MSG_TYPE_PROPOSAL = 32;
 }
 ```
 
-There are two types of votes:
-a _prevote_ has `vote.Type == 1` and
-a _precommit_ has `vote.Type == 2`.
+## CanonicalVote
```

Comment:
I don't understand how this is linked to the rest of the document.

Reply:
Canonical Vote is used for signing (privval). It can be seen as implementation specific, but the rust client has implemented it due to its use in tmkms.

Reply:
I see. My point is that if I read the document from top to bottom, at this point I have no idea what a Canonical Vote is. At the beginning of the document there is a bullet list with data types, but it doesn't show there. Also it is not mentioned. So the role of this data type is just not clearly stated here, I guess.